### PR TITLE
Add changelog for #3246 and #3429

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Add a changelog entry for #3246 and #3429.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
No.

Checklist
---------
- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- ~~[ ] I have added tests for all code changes.~~
- ~~[ ] I have added documentation for relevant changes (in the manual).~~
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- ~~[ ] I have run `gofmt` on the code in all commits.~~
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: changelog/unreleased/issue-2202 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Enhancement: Add upload checksum for azure, gs, s3 and swift backends
+
+Up to now only the B2 backend and partially the Swift backend verified the
+integrity of uploaded files. The verification works by informing the backend
+about the expected hash of the uploaded data. This allows the backend to verify
```
Comment (greatroar) [inline changelog/unreleased/issue-2202:5]: Maybe stress that the hash is calculated on the *encrypted* data?

Comment (fd0) [reply changelog/unreleased/issue-2202:5]: Isn't that clear from the context (all data stored in the backend is encrypted)?

**Review by fd0 — APPROVED** [review]
LGTM, just waiting for Michael to comment

File: changelog/unreleased/issue-2202 (hunk 1 above)
Comment (MichaelEischer) [reply changelog/unreleased/issue-2202]: I've slightly tweaked the wording to mention that the uploaded files are encrypted. And `uploaded data` is now replaced with `uploaded files`.

**Pull request merged**